.github/workflows/test.yml

```diff
@@ -45,52 +61,82 @@ jobs:
+          EOF
       - name: Bazel build
         # timestamps are no longer being added here, the GitHub logs include timestamps (Use
         # 'Show timestamps' on the web interface)
         run: |
-            bazelisk build --disk_cache=~/bazel-disk-cache --remote_cache=https://bazel:${{ secrets.BAZEL_CACHE_KEY }}@bazel-remote-cache.devprod.cloudflare.dev --verbose_failures //...
-      - name: Bazel tests
-        # Run tests and check that there are no .bazelrc issues that prevent shutdown.
+          bazelisk build ${{ matrix.config.bazel-args }} --disk_cache=~/bazel-disk-cache --remote_cache=https://bazel:${{ secrets.BAZEL_CACHE_KEY }}@bazel-remote-cache.devprod.cloudflare.dev --verbose_failures //...
+      - name: Bazel test
         run: |
-            bazelisk test --disk_cache=~/bazel-disk-cache --remote_cache=https://bazel:${{ secrets.BAZEL_CACHE_KEY }}@bazel-remote-cache.devprod.cloudflare.dev --verbose_failures --keep_going --test_output=errors //...
-            bazelisk shutdown
+          bazelisk test ${{ matrix.config.bazel-args }} --disk_cache=~/bazel-disk-cache --remote_cache=https://bazel:${{ secrets.BAZEL_CACHE_KEY }}@bazel-remote-cache.devprod.cloudflare.dev --keep_going --verbose_failures --test_output=errors //...
+      - name: Bazel shutdown
+        # Check that there are no .bazelrc issues that prevent shutdown.
+        run: bazelisk shutdown
       - name: Report disk usage
         if: always()
         shell: bash
         run: |
-            BAZEL_OUTPUT_BASE=$(bazel info --ui_event_filters=-WARNING output_base)
-            BAZEL_REPOSITORY_CACHE=$(bazel info --ui_event_filters=-WARNING repository_cache)
-            echo "Bazel cache usage statistics"
-            du -hs -t 1 ~/bazel-disk-cache/* $BAZEL_REPOSITORY_CACHE
-            echo "Bazel output usage statistics"
-            du -hs -t 1 $BAZEL_OUTPUT_BASE
-            echo "Workspace usage statistics"
-            du -hs -t 1 $GITHUB_WORKSPACE
-
+          BAZEL_OUTPUT_BASE=$(bazel info --ui_event_filters=-WARNING output_base)
+          BAZEL_REPOSITORY_CACHE=$(bazel info --ui_event_filters=-WARNING repository_cache)
+          echo "Bazel cache usage statistics"
+          du -hs -t 1 ~/bazel-disk-cache/* $BAZEL_REPOSITORY_CACHE
+          echo "Bazel output usage statistics"
+          du -hs -t 1 $BAZEL_OUTPUT_BASE
+          echo "Workspace usage statistics"
+          du -hs -t 1 $GITHUB_WORKSPACE
+      - name: Drop large Bazel cache files
+        # Github has a nominal 10GB of storage for all cached builds associated with a project.
+        # Drop large files (>100MB) in our cache to improve shared build cache efficiency. This is
+        # particularly helpful for asan and debug builds that produce larger executables. Also
+        # the process of saving the Bazel disk cache generates a tarball on the runners disk, and
+        # it is possible to run out of storage in that process (does not fail the workflow).
+        shell: bash
```

Comment:
This should also be `if: always()` – curious to see how much this saves for the windows-debug cache size

Reply:
Coming in the next push. Thanks.